var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -702,33 +693,36 @@ def configure_args(self):
             ])
 
         # Singularity container support
-        if spec.satisfies('+singularity @:4.9'):
+        if spec.satisfies('+singularity'):
             singularity_opt = '--with-singularity={0}'.format(
                 spec['singularity'].prefix)
             config_args.append(singularity_opt)
         # Lustre filesystem support
         if spec.satisfies('+lustre'):
             lustre_opt = '--with-lustre={0}'.format(spec['lustre'].prefix)
             config_args.append(lustre_opt)
-        # external libevent
-        if spec.satisfies('@4.0.0:') or spec.satisfies('+pmix'):
+        # External libevent/pmix
+        if spec.satisfies('@5:') or spec.satisfies('+pmix'):
+            config_args.append('--with-pmix={0}'.format(spec['pmix'].prefix))
             config_args.append('--with-libevent={0}'.format(spec['libevent'].prefix))
+        elif spec.satisfies('~pmix'):
+            config_args.append('--without-pmix')
+
         # Hwloc support
-        if '~internal-hwloc' in spec and spec.satisfies('@1.5.2:'):
+        if 'hwloc' in spec:
```

Comment:
Maybe explicit `^hwloc` to make it more clear it applies to a dependency?

Reply:
I wonder if the `internal-hwloc` variant should be deleted... seems like it's asking for a conflict, and as a bonus the variant name uses the ambiguous `-` rather than `_`.